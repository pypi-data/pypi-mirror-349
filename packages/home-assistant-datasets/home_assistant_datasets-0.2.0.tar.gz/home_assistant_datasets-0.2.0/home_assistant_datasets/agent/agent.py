"""Conversation agent module."""

from abc import ABC, abstractmethod
from typing import Any

from homeassistant.core import HomeAssistant


class ConversationAgent(ABC):
    """A simple client library for calling a conversation agent."""

    @abstractmethod
    async def async_process(self, hass: HomeAssistant, text: str) -> str:
        """Process a text input and return the response."""

    @abstractmethod
    def trace_context(self) -> dict[str, Any]:
        """Record any relevant trace context captured during the requests."""
