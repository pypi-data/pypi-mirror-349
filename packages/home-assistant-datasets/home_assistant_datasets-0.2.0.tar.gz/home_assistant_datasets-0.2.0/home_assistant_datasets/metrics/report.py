"""Common classes for reporting."""

from abc import ABC, abstractmethod


from . import ScrapeRecord, TaskResult


__all__ = [
    "ScrapeRecordWriter",
    "TaskResultWriter",
]


class ScrapeRecordWriter(ABC):
    """Base class for eval output."""

    @abstractmethod
    def start(self) -> None:
        """Write the output start."""

    @abstractmethod
    def row(self, scrape: ScrapeRecord) -> None:
        """Write an output row."""

    @abstractmethod
    def finish(self) -> None:
        """Write the output summary."""


class TaskResultWriter(ABC):
    """Base class for eval output."""

    @abstractmethod
    def start(self) -> None:
        """Write the output start."""

    @abstractmethod
    def row(self, scrape: ScrapeRecord, result: TaskResult) -> None:
        """Write an output row."""

    @abstractmethod
    def finish(self) -> None:
        """Write the output summary."""
