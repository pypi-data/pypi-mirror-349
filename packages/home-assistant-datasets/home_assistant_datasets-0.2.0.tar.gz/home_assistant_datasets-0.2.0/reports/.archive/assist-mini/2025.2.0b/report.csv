task_id,model_id,label,category,task_prefix,text,response,tool_call,entity_diff
"dom1_pl_lights_lights-dining_room_light_off","qwen2.5","Good","light","dom1_pl_lights_lights","Dining room light off","The Dining Room light is now off.","{'tool_name': 'HassTurnOff', 'tool_args': {'area': 'Dining Room', 'domain': ['light'], 'name': 'Dining Room Light'}}","{}"
"dom1_pl_lights_lights-kitchen_light_off","qwen2.5","Good","light","dom1_pl_lights_lights","Kitchen light off","The Kitchen light has been turned off.","{'tool_name': 'HassTurnOff', 'tool_args': {'area': 'Kitchen', 'domain': ['light'], 'name': 'Kitchen Light'}}","{}"
"dom1_pl_lights_lights-please_turn_on_the_kitchen_light","qwen2.5","Good","light","dom1_pl_lights_lights","Please turn on the kitchen light","The kitchen light has been turned on.","{'tool_name': 'HassTurnOn', 'tool_args': {'area': 'Kitchen', 'domain': ['light'], 'name': 'Kitchen Light'}}","{}"
"dom1_pl_lights_lights-turn_on_the_backyard_light","qwen2.5","Good","light","dom1_pl_lights_lights","Turn on the Backyard light","The Backyard light has been turned on.","{'tool_name': 'HassTurnOn', 'tool_args': {'area': 'Backyard', 'domain': ['light'], 'name': 'Garden Light'}}","{}"
"dom1_pl_lights_lights-turn_on_the_living_room_light","qwen2.5","Good","light","dom1_pl_lights_lights","Turn on the living room light","The Living Room Light has been turned on.","{'tool_name': 'HassTurnOn', 'tool_args': {'area': 'Living Room', 'domain': ['light'], 'name': 'Living Room Light'}}","{}"
"dom1_pl_todo_todo-add_clean_the_kitchen_to_my_todo_list","qwen2.5","Good","todo","dom1_pl_todo_todo","Add clean the kitchen to my todo list","I've added 'clean the kitchen' to your todo list under Personal Tasks.","{'tool_name': 'HassListAddItem', 'tool_args': {'item': 'clean the kitchen', 'name': 'Personal Tasks'}}","{}"
"dom1_pl_todo_todo-add_history_homework_to_my_personal_tasks","qwen2.5","Good","todo","dom1_pl_todo_todo","Add history homework to my personal tasks","I have added 'history homework' to your personal tasks.","{'tool_name': 'HassListAddItem', 'tool_args': {'item': 'history homework', 'name': 'Personal Tasks'}}","{}"
"dom1_pl_todo_todo-add_history_homework_to_my_tasks","qwen2.5","Good","todo","dom1_pl_todo_todo","Add history homework to my tasks","I have added 'history homework' to your tasks.","{'tool_name': 'HassListAddItem', 'tool_args': {'item': 'history homework', 'name': 'Personal Tasks'}}","{}"
"dom1_pl_todo_todo-put_history_homework_on_personal_tasks_todo_list","qwen2.5","Good","todo","dom1_pl_todo_todo","Put history homework on personal tasks todo list","I have added 'History homework' to your personal tasks todo list.","{'tool_name': 'HassListAddItem', 'tool_args': {'item': 'History homework', 'name': 'Personal Tasks'}}","{}"
"home1_us_cover_garage_cover_garage-close_the_garage_door","qwen2.5","Good","cover","home1_us_cover_garage_cover_garage","Close the garage door","The garage door has been closed.","{'tool_name': 'HassTurnOff', 'tool_args': {'area': 'Garage', 'device_class': ['garage'], 'domain': ['cover'], 'name': 'Garage Door Opener'}}","{}"
"home1_us_cover_garage_cover_garage-open_the_garage_door","qwen2.5","Good","cover","home1_us_cover_garage_cover_garage","Open the garage door","The garage door is now opening.","{'tool_name': 'HassTurnOn', 'tool_args': {'area': 'Garage', 'domain': ['cover'], 'name': 'Garage Door Opener'}}","{}"
"home1_us_cover_garage_cover_garage-please_close_the_garage_door","qwen2.5","Good","cover","home1_us_cover_garage_cover_garage","Please close the garage door","The garage door has been closed.","{'tool_name': 'HassTurnOff', 'tool_args': {'area': 'Garage', 'domain': ['cover']}}","{}"
"home1_us_cover_garage_cover_garage-please_open_the_garage_door","qwen2.5","Good","cover","home1_us_cover_garage_cover_garage","Please open the garage door","The garage door has been opened.","{'tool_name': 'HassTurnOn', 'tool_args': {'area': 'Garage', 'domain': ['cover'], 'name': 'Garage Door Opener'}}","{}"
"home1_us_lock_smart_lock-lock_smart_lock","qwen2.5","Good","lock","home1_us_lock_smart_lock","Lock smart lock","Smart lock in the Entry area has been locked.","{'tool_name': 'HassTurnOn', 'tool_args': {'area': 'Entry', 'domain': 'lock', 'name': 'Smart Lock'}}","{}"
"home1_us_lock_smart_lock-lock_the_entry_lock","qwen2.5","Good","lock","home1_us_lock_smart_lock","Lock the entry lock","The entry lock has been locked.","{'tool_name': 'HassTurnOn', 'tool_args': {'area': 'Entry', 'domain': ['lock'], 'name': 'Smart Lock'}}","{}"
"home1_us_lock_smart_lock-unlock_smart_lock","qwen2.5","Bad","lock","home1_us_lock_smart_lock","Unlock smart lock","The smart lock in the Entry area has been unlocked.","{'tool_name': 'HassTurnOn', 'tool_args': {'area': 'Entry', 'domain': 'lock', 'name': 'Smart Lock'}}","{'lock.smart_lock': {'expected': {'state': 'unlocked'}, 'got': {'state': 'locked'}}}"
"home1_us_lock_smart_lock-unlock_the_entry_lock","qwen2.5","Bad","lock","home1_us_lock_smart_lock","Unlock the entry lock","The entry lock has been unlocked.","{'tool_name': 'HassTurnOn', 'tool_args': {'area': 'Entry', 'domain': 'lock', 'name': 'Smart Lock'}}","{'lock.smart_lock': {'expected': {'state': 'unlocked'}, 'got': {'state': 'locked'}}}"
"home1_us_vacuum_vacuum-return_roborock_downstairs_to_base","qwen2.5","Good","vacuum","home1_us_vacuum_vacuum","Return Roborock Downstairs to base","I have returned Roborock Downstairs to its base.","{'tool_name': 'HassVacuumReturnToBase', 'tool_args': {'area': 'Living Room', 'domain': ['vacuum'], 'floor': 'Downstairs', 'name': 'Roborock Downstairs'}}","{}"
"home1_us_vacuum_vacuum-return_vacuum_in_the_living_room_to_base","qwen2.5","Good","vacuum","home1_us_vacuum_vacuum","Return vacuum in the living room to base","The vacuum in the living room has been returned to its base.","{'tool_name': 'HassVacuumReturnToBase', 'tool_args': {'area': 'Living Room', 'domain': ['vacuum']}}","{}"
"home1_us_vacuum_vacuum-start_roborock_downstairs_vacuum","qwen2.5","Good","vacuum","home1_us_vacuum_vacuum","Start Roborock Downstairs vacuum","The Roborock Downstairs vacuum has started cleaning in the Living Room.","{'tool_name': 'HassVacuumStart', 'tool_args': {'area': 'Living Room', 'domain': ['vacuum'], 'name': 'Roborock Downstairs'}}","{}"
"home1_us_vacuum_vacuum-start_vacuum_in_the_living_room","qwen2.5","Good","vacuum","home1_us_vacuum_vacuum","Start vacuum in the living room","The vacuum in the living room has been started. It will clean starting from the Roborock Downstairs device.","{'tool_name': 'HassVacuumStart', 'tool_args': {'area': 'Living Room', 'domain': ['vacuum']}}","{}"
"home2_ru_valve_water_valve-close_the_front_yard_valve","qwen2.5","Good","valve","home2_ru_valve_water_valve","close the front yard valve","The front yard valve has been turned off.","{'tool_name': 'HassTurnOff', 'tool_args': {'area': 'Front yard', 'domain': 'valve', 'name': 'Irrigation Valve'}}","{}"
"home2_ru_valve_water_valve-close_the_irrigation_valve","qwen2.5","Good","valve","home2_ru_valve_water_valve","close the irrigation valve","The irrigation valve has been successfully closed.","{'tool_name': 'HassTurnOff', 'tool_args': {'area': 'Front yard', 'domain': 'valve', 'name': 'Irrigation Valve'}}","{}"
"home2_ru_valve_water_valve-open_the_front_yard_valve","qwen2.5","Good","valve","home2_ru_valve_water_valve","open the front yard valve","The front yard valve has been turned on. However, there was an issue with the Irrigation Valve itself and also the meter reading sensor. Please check them manually.","{'tool_name': 'HassTurnOn', 'tool_args': {'area': 'Front yard', 'device': 'Irrigation Valve'}}","{}"
"home2_ru_valve_water_valve-open_the_irrigation_valve","qwen2.5","Good","valve","home2_ru_valve_water_valve","open the irrigation valve","The irrigation valve in the front yard has been opened.","{'tool_name': 'HassTurnOn', 'tool_args': {'area': 'Front yard', 'domain': ['valve'], 'name': 'Irrigation Valve'}}","{}"
"home2_ru_valve_water_valve-set_the_irrigation_valve_to_50","qwen2.5","Bad","valve","home2_ru_valve_water_valve","set the irrigation valve to 50%","The position has been set to 50% for the Irrigation Valve in the Front yard.","{'tool_name': 'HassSetPosition', 'tool_args': {'area': 'Front yard', 'device_class': ['valve'], 'domain': ['valve'], 'name': 'Irrigation Valve', 'position': 50}}","{'valve.irrigation_valve': {'expected': {'current_position': 50}, 'got': {'current_position': 100}}}"
"home5_cn_fan_fan-turn_off_the_bedroom_1_fan","qwen2.5","Good","fan","home5_cn_fan_fan","Turn off the bedroom 1 fan","The bedroom 1 fan has been turned off.","{'tool_name': 'HassTurnOff', 'tool_args': {'area': 'Bedroom 1', 'domain': 'fan', 'name': 'Bedroom Fan'}}","{}"
"home5_cn_fan_fan-turn_off_the_bedroom_fan","qwen2.5","Good","fan","home5_cn_fan_fan","Turn off the bedroom fan","The bedroom fan has been turned off.","{'tool_name': 'HassTurnOff', 'tool_args': {'domain': 'fan', 'name': 'Bedroom Fan'}}","{}"
"home5_cn_fan_fan-turn_on_the_bedroom_1_fan","qwen2.5","Good","fan","home5_cn_fan_fan","Turn on the bedroom 1 fan","The Bedroom 1 fan has been turned on.","{'tool_name': 'HassTurnOn', 'tool_args': {'area': 'Bedroom 1', 'domain': 'fan', 'name': 'Bedroom Fan'}}","{}"
"home5_cn_fan_fan-turn_on_the_bedroom_fan","qwen2.5","Good","fan","home5_cn_fan_fan","Turn on the bedroom fan","The bedroom fan has been turned on.","{'tool_name': 'HassTurnOn', 'tool_args': {'domain': 'fan', 'name': 'Bedroom Fan'}}","{}"
"home5_cn_light_light-set_the_living_room_light_to_50_brightness","qwen2.5","Bad","light","home5_cn_light_light","Set the living room light to 50% brightness","The living room light is now set to 50% brightness.","{'tool_name': 'HassLightSet', 'tool_args': {'area': 'Living Room', 'brightness': 128, 'domain': ['light'], 'name': 'Living Room Light'}}","{'light.living_room_light': {'expected': {'brightness': 128}, 'got': {'brightness': 255}}}"
"home7_dk_cover_curtain_cover_curtain-close_the_living_room_curtains","qwen2.5","Good","cover","home7_dk_cover_curtain_cover_curtain","Close the living room curtains","The living room curtains have been closed.","{'tool_name': 'HassTurnOff', 'tool_args': {'area': 'Living Room', 'domain': 'cover', 'name': 'Smart Curtain'}}","{}"
"home7_dk_cover_curtain_cover_curtain-close_the_living_room_smart_curtain","qwen2.5","Good","cover","home7_dk_cover_curtain_cover_curtain","Close the living room smart curtain","The living room smart curtain has been closed.","{'tool_name': 'HassTurnOff', 'tool_args': {'area': 'Living Room', 'domain': 'cover', 'name': 'Smart Curtain'}}","{}"
"home7_dk_cover_curtain_cover_curtain-close_the_smart_cutains","qwen2.5","Good","cover","home7_dk_cover_curtain_cover_curtain","Close the smart cutains","The smart curtain in the Living Room has been closed.","{'tool_name': 'HassTurnOff', 'tool_args': {'area': 'Living Room', 'domain': 'cover', 'name': 'Smart Curtain'}}","{}"
"home7_dk_cover_curtain_cover_curtain-open_the_living_room_curtains","qwen2.5","Good","cover","home7_dk_cover_curtain_cover_curtain","Open the living room curtains","The living room curtains have been opened.","{'tool_name': 'HassTurnOn', 'tool_args': {'area': 'Living Room', 'domain': 'cover', 'name': 'Smart Curtain'}}","{}"
"home7_dk_cover_curtain_cover_curtain-open_the_living_room_smart_curtain","qwen2.5","Good","cover","home7_dk_cover_curtain_cover_curtain","Open the living room smart curtain","The living room smart curtain has been opened.","{'tool_name': 'HassTurnOn', 'tool_args': {'area': 'Living Room', 'domain': 'cover', 'name': 'Smart Curtain'}}","{}"
"home7_dk_cover_curtain_cover_curtain-open_the_smart_curtain","qwen2.5","Good","cover","home7_dk_cover_curtain_cover_curtain","Open the smart curtain","The smart curtain in the Living Room has been opened.","{'tool_name': 'HassTurnOn', 'tool_args': {'area': 'Living Room', 'domain': ['cover'], 'name': 'Smart Curtain'}}","{}"
"home7_dk_cover_curtain_cover_curtain-set_the_living_room_curtains_to_50","qwen2.5","Good","cover","home7_dk_cover_curtain_cover_curtain","Set the living room curtains to 50%","The living room curtains are now at 50%.","{'tool_name': 'HassSetPosition', 'tool_args': {'area': 'Living Room', 'domain': 'cover', 'name': 'Smart Curtain', 'position': 50}}","{}"
"home7_dk_cover_curtain_cover_curtain-set_the_smart_curtain_to_50","qwen2.5","Good","cover","home7_dk_cover_curtain_cover_curtain","Set the smart curtain to 50%","The smart curtain in the Living Room has been set to 50%.","{'tool_name': 'HassSetPosition', 'tool_args': {'area': 'Living Room', 'domain': ['cover'], 'name': 'Smart Curtain', 'position': 50}}","{}"
"home7_dk_media_player_media_player-mute_the_outdoor_speakers","qwen2.5","Bad","media-player","home7_dk_media_player_media_player","Mute the outdoor speakers","The outdoor speakers have been turned off. They will no longer play any media until they are turned back on.","{'tool_name': 'HassSetVolume', 'tool_args': {'area': 'Rooftop Terrace', 'device_class': ['speaker'], 'domain': ['media_player']}}","{'media_player.outdoor_speakers': {'expected': {'volume_level': 0.0, 'media_track': 1, 'state': 'playing'}, 'got': {'volume_level': None, 'media_track': None, 'state': 'off'}}}"
"home7_dk_media_player_media_player-pause_outdoor_speakers","qwen2.5","Good","media-player","home7_dk_media_player_media_player","Pause Outdoor Speakers","The Outdoor Speakers on the Rooftop Terrace have been paused.","{'tool_name': 'HassMediaPause', 'tool_args': {'area': 'Rooftop Terrace', 'domain': 'media_player', 'name': 'Outdoor Speakers'}}","{}"
"home7_dk_media_player_media_player-pause_the_rooftop_terrace_music","qwen2.5","Good","media-player","home7_dk_media_player_media_player","Pause the Rooftop Terrace music","The music on the Rooftop Terrace has been paused.","{'tool_name': 'HassMediaPause', 'tool_args': {'area': 'Rooftop Terrace', 'domain': 'media_player'}}","{}"
"home7_dk_media_player_media_player-resume_outdoor_speakers","qwen2.5","Good","media-player","home7_dk_media_player_media_player","Resume Outdoor Speakers","The Outdoor Speakers in the Rooftop Terrace have been resumed.","{'tool_name': 'HassMediaUnpause', 'tool_args': {'area': 'Rooftop Terrace', 'domain': ['media_player'], 'name': 'Outdoor Speakers'}}","{}"
"home7_dk_media_player_media_player-set_outdoor_speakers_volume_to_50","qwen2.5","Bad","media-player","home7_dk_media_player_media_player","Set outdoor speakers volume to 50%","The volume for the outdoor speakers has been set to 50%.","{'tool_name': 'HassSetVolume', 'tool_args': {'area': 'Rooftop Terrace', 'device_class': ['speaker'], 'domain': ['media_player'], 'volume_level': 0.5}}","{'media_player.outdoor_speakers': {'expected': {'volume_level': 0.5}, 'got': {'volume_level': 0.0}}}"
"home7_dk_media_player_media_player-skip_song_on_outdoor_speakers","qwen2.5","Good","media-player","home7_dk_media_player_media_player","Skip song on outdoor speakers","The next song is now playing on the outdoor speakers at Rooftop Terrace.","{'tool_name': 'HassMediaNext', 'tool_args': {'area': 'Rooftop Terrace', 'domain': 'media_player', 'name': 'Outdoor Speakers'}}","{}"
"home7_dk_media_player_media_player-skip_to_the_next_track_on_the_outdoor_speakers","qwen2.5","Good","media-player","home7_dk_media_player_media_player","Skip to the next track on the outdoor speakers","The next track has been skipped on the outdoor speakers in the Rooftop Terrace area.","{'tool_name': 'HassMediaNext', 'tool_args': {'area': 'Rooftop Terrace', 'domain': ['media_player'], 'name': 'Outdoor Speakers'}}","{}"
"home7_dk_media_player_media_player-stop_the_outdoor_speakers","qwen2.5","Good","media-player","home7_dk_media_player_media_player","Stop the Outdoor Speakers","The Outdoor Speakers have been paused.","{'tool_name': 'HassMediaPause', 'tool_args': {'area': 'Rooftop Terrace', 'domain': 'media_player', 'name': 'Outdoor Speakers'}}","{}"
"home7_dk_media_player_media_player-turn_the_volume_down_to_50","qwen2.5","Bad","media-player","home7_dk_media_player_media_player","Turn the volume down to 50%","The volume for the Outdoor Speakers has been set to 50%.","{'tool_name': 'HassSetVolume', 'tool_args': {'domain': 'media_player', 'name': 'Outdoor Speakers', 'volume_level': 0.5}}","{'media_player.outdoor_speakers': {'expected': {'volume_level': 0.5}, 'got': {'volume_level': 0.0}}}"
"home7_dk_media_player_media_player-unpause_the_rooftop_terrace_music","qwen2.5","Good","media-player","home7_dk_media_player_media_player","Unpause the Rooftop Terrace music","The music on the Outdoor Speakers at Rooftop Terrace has been unpause.","{'tool_name': 'HassMediaUnpause', 'tool_args': {'area': 'Rooftop Terrace', 'domain': 'media_player', 'name': 'Outdoor Speakers'}}","{}"
