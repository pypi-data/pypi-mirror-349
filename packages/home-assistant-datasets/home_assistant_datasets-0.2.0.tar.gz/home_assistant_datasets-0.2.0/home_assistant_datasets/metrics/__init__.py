"""Module for computing eval related metrics and reporting."""

from typing import Any
from dataclasses import field, dataclass

from home_assistant_datasets.agent.trace_events import TokenStats


__all__ = [
    "ScrapeRecord",
    "TaskResult",
    "GOOD_LABEL",
    "BAD_LABEL",
    "report",
    "report_suite",
    "accuracy",
    "token_stats",
    "report_csv",
]


GOOD_LABEL = "Good"
BAD_LABEL = "Bad"


@dataclass
class ScrapeRecord:
    """Identifier for a particular model scrape/collection task."""

    uuid: str
    """The unique identifier for the model output/prediction."""

    task_id: str
    """An identifier for the task in the dataset."""

    model_id: str
    """The model identifier used to generate the prediction."""

    context: dict[str, Any] = field(default_factory=dict)
    """Additional context/detail from runtime of evaluating the prediction."""

    token_stats: TokenStats | None = None
    """Token statistics for the model output."""

    extra_data: dict[str, Any] | None = None
    """Additional task specific data to include in the detailed eval report."""

    def as_report_data(self) -> dict[str, Any]:
        """Export the record for reporting."""
        return {
            "task_id": self.task_id,
            "model_id": self.model_id,
            **(self.extra_data if self.extra_data is not None else {}),
        }


@dataclass
class TaskResult:
    """Result of a specific measurement task for a scrape record."""

    task_name: str
    """The name of the particular test for this test e.g. "accuracy"."""

    passed: bool
    """If true, the test passed. If false, it failed."""

    detail: str | None = None
    """Additional detail about the test failure."""

    skipped: bool = False
    """Entirely ignore this pass/fail result.."""

    def as_report_data(self) -> dict[str, Any] | None:
        """Export the record for reporting."""
        if self.skipped:
            return None
        return {
            "task_name": self.task_name,
            "label": GOOD_LABEL if self.passed else BAD_LABEL,
            "details": self.detail,
        }
