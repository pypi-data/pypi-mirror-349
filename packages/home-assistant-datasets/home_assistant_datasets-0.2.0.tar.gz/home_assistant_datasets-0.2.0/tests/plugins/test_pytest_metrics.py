"""Tests for the eval report generation pytest plugin."""

import tempfile
import pathlib

import pytest
from syrupy import SnapshotAssertion

PLUGINS = ["home_assistant_datasets.plugins.pytest_metrics"]
PYTEST_INI = """
[pytest]
asyncio_mode = auto
asyncio_default_fixture_loop_scope = function
"""

MODEL_OUTPUT_YAML = """---
uuid: e489d6b0-0f5f-4c49-aced-b9abe1c5104f
task_id: urban_loft_au_light-how_many_lights_are_currently_on_please_respond_with_a_number
model_id: example-model
category: light
task:
  input_text: How many lights are currently on? Please respond with a number.
  expect_changes: {}
  expect_response:
  - 4
response: |-
  Based on the current state of the devices in the home, I can provide you with the number of lights that are currently on.

  4
context:
  unexpected_states: {}
  conversation_trace: {}
  tries: 0
  duration_ms: 3582.574
"""

TEST_FILE_CONTENTS_FORMAT = """
    from dataclasses import dataclass
    from typing import Any

    import pytest

    from home_assistant_datasets.agent.trace_events import TokenStats
    from home_assistant_datasets.metrics.report import ScrapeRecord

    @pytest.mark.eval_model_outputs
    def test_success_1(scrape_record: Any) -> None:
        assert True

    @pytest.mark.eval_model_outputs
    def test_success_2() -> None:
        assert True

    @pytest.mark.eval_model_outputs
    def test_fail() -> None:
        assert False, "Failed test case"

    @pytest.mark.eval_model_outputs
    def test_skip() -> None:
        pytest.skip()

    @pytest.mark.eval_model_outputs
    @pytest.mark.parametrize(
        "exc",
        [
            None,
            ValueError("This is a test"),
            TimeoutError("This is a test"),
            TypeError("This is a test"),
            AssertionError("This is a test"),
        ],
    )
    def test_exception_repr(exc: Exception | None) -> None:
        if exc is not None:
            raise exc
"""


@pytest.fixture(name="tmpdir")
def tmpdir_fixture() -> str:
    with tempfile.TemporaryDirectory() as tmpdir:
        yield str(tmpdir)


REPORT_OUTPUT_PREFIX = "Generated eval report:"


def parse_report_filenames(tmpdir: str, stdout_list: str) -> list[pathlib.Path]:
    """Parse report filenames from test result stdout.

    Parse the output lines for "Generated eval report" filenames.
    """
    report_files: list[pathlib.Path] = []
    for line in stdout_list:
        if REPORT_OUTPUT_PREFIX not in line or tmpdir not in line:
            continue
        line = line[line.find(tmpdir) :]
        line = line[: line.find(" ")]
        report_files.append(pathlib.Path(line))
    return report_files


def test_verify_report_output_files(
    pytester: pytest.Pytester, tmpdir: str, snapshot: SnapshotAssertion
) -> None:
    """Exercise the report plugin."""

    pytester.makepyfile(TEST_FILE_CONTENTS_FORMAT.format(tmpdir=tmpdir))
    pytester.makeini(PYTEST_INI)

    # Prepare a fake scraped model output file in the model output directory.
    model_output_path = pathlib.Path(tmpdir)
    (model_output_path / "model-1").mkdir()
    (model_output_path / "model-1" / "model-1-output.yaml").write_text(
        MODEL_OUTPUT_YAML
    )

    pytester.plugins.extend(PLUGINS)
    result = pytester.runpytest_subprocess(
        "--model_output_dir",
        tmpdir,
    )

    # Parse the output lines for "Generated eval report" filenames
    report_files = parse_report_filenames(tmpdir, result.stdout.lines)

    # Snapshot the contents of the generated reports
    assert [
        (report_file.name, report_file.read_text()) for report_file in report_files
    ] == snapshot
