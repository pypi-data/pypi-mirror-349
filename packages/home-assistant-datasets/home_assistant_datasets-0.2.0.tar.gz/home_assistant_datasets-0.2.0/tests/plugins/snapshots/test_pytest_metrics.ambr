# serializer version: 1
# name: test_verify_report_output_files
  list([
    tuple(
      'reports-token-stats.yaml',
      '''
        --- []
        
  
      ''',
    ),
    tuple(
      'reports.yaml',
      '''
        ---
        - model_id: example-model
          good_percent: 37.5%
          confidence_interval: 33.5%
          good: 3
          total: 8
        
  
      ''',
    ),
    tuple(
      'reports-by-model-test-name.yaml',
      '''
        ---
        - model_id-task_name: example-model-test_verify_report_output_files0-test_exception_repr
          good_percent: 20.0%
          confidence_interval: 35.1%
          good: 1
          total: 5
        - model_id-task_name: example-model-test_verify_report_output_files0-test_fail
          good_percent: 0.0%
          confidence_interval: 0.0%
          good: 0
          total: 1
        - model_id-task_name: example-model-test_verify_report_output_files0-test_skip
          good_percent: 0.0%
          confidence_interval: 0.0%
          good: 0
          total: 0
        - model_id-task_name: example-model-test_verify_report_output_files0-test_success_1
          good_percent: 100.0%
          confidence_interval: 0.0%
          good: 1
          total: 1
        - model_id-task_name: example-model-test_verify_report_output_files0-test_success_2
          good_percent: 100.0%
          confidence_interval: 0.0%
          good: 1
          total: 1
        
  
      ''',
    ),
    tuple(
      'reports-by-model-category.yaml',
      '''
        ---
        - model_id-category: example-model-light
          good_percent: 37.5%
          confidence_interval: 33.5%
          good: 3
          total: 8
        
  
      ''',
    ),
    tuple(
      'reports-by-test-name.yaml',
      '''
        ---
        - task_name: test_verify_report_output_files0-test_exception_repr
          good_percent: 20.0%
          confidence_interval: 35.1%
          good: 1
          total: 5
        - task_name: test_verify_report_output_files0-test_fail
          good_percent: 0.0%
          confidence_interval: 0.0%
          good: 0
          total: 1
        - task_name: test_verify_report_output_files0-test_skip
          good_percent: 0.0%
          confidence_interval: 0.0%
          good: 0
          total: 0
        - task_name: test_verify_report_output_files0-test_success_1
          good_percent: 100.0%
          confidence_interval: 0.0%
          good: 1
          total: 1
        - task_name: test_verify_report_output_files0-test_success_2
          good_percent: 100.0%
          confidence_interval: 0.0%
          good: 1
          total: 1
        
  
      ''',
    ),
    tuple(
      'reports-by-task-id.yaml',
      '''
        ---
        - task_id: urban_loft_au_light-how_many_lights_are_currently_on_please_respond_with_a_number
          good_percent: 37.5%
          confidence_interval: 33.5%
          good: 3
          total: 8
        
  
      ''',
    ),
    tuple(
      'reports-by-category.yaml',
      '''
        ---
        - category: light
          good_percent: 37.5%
          confidence_interval: 33.5%
          good: 3
          total: 8
        
  
      ''',
    ),
    tuple(
      'report.csv',
      '''
        task_id,model_id,category,text,tool_call,response,task_name,label,details
        "urban_loft_au_light-how_many_lights_are_currently_on_please_respond_with_a_number","example-model","light","How many lights are currently on? Please respond with a number.","","Based on the current state of the devices in the home, I can provide you with the number of lights that are currently on.\n\n4","test_verify_report_output_files0-test_success_1","Good",""
        "urban_loft_au_light-how_many_lights_are_currently_on_please_respond_with_a_number","example-model","light","How many lights are currently on? Please respond with a number.","","Based on the current state of the devices in the home, I can provide you with the number of lights that are currently on.\n\n4","test_verify_report_output_files0-test_success_2","Good",""
        "urban_loft_au_light-how_many_lights_are_currently_on_please_respond_with_a_number","example-model","light","How many lights are currently on? Please respond with a number.","","Based on the current state of the devices in the home, I can provide you with the number of lights that are currently on.\n\n4","test_verify_report_output_files0-test_fail","Bad","AssertionError: Failed test case,assert False"
        "urban_loft_au_light-how_many_lights_are_currently_on_please_respond_with_a_number","example-model","light","How many lights are currently on? Please respond with a number.","","Based on the current state of the devices in the home, I can provide you with the number of lights that are currently on.\n\n4","test_verify_report_output_files0-test_exception_repr","Good",""
        "urban_loft_au_light-how_many_lights_are_currently_on_please_respond_with_a_number","example-model","light","How many lights are currently on? Please respond with a number.","","Based on the current state of the devices in the home, I can provide you with the number of lights that are currently on.\n\n4","test_verify_report_output_files0-test_exception_repr","Bad","ValueError: This is a test"
        "urban_loft_au_light-how_many_lights_are_currently_on_please_respond_with_a_number","example-model","light","How many lights are currently on? Please respond with a number.","","Based on the current state of the devices in the home, I can provide you with the number of lights that are currently on.\n\n4","test_verify_report_output_files0-test_exception_repr","Bad","TimeoutError: This is a test"
        "urban_loft_au_light-how_many_lights_are_currently_on_please_respond_with_a_number","example-model","light","How many lights are currently on? Please respond with a number.","","Based on the current state of the devices in the home, I can provide you with the number of lights that are currently on.\n\n4","test_verify_report_output_files0-test_exception_repr","Bad","TypeError: This is a test"
        "urban_loft_au_light-how_many_lights_are_currently_on_please_respond_with_a_number","example-model","light","How many lights are currently on? Please respond with a number.","","Based on the current state of the devices in the home, I can provide you with the number of lights that are currently on.\n\n4","test_verify_report_output_files0-test_exception_repr","Bad","AssertionError: This is a test"
  
      ''',
    ),
  ])
# ---
